spinc/cmd/find.go

```diff
@@ -0,0 +1,257 @@
+		until, err = time.Parse(findTimeFmtStr, args["until"])
+		if err != nil {
+			return fmt.Errorf("Invalid time %s, expected form '%s'", args["until"], findTimeFmt)
+		}
+	}
+
+	var limit uint
+	if args["limit"] == "" {
+		limit = findLimitDefault
+	} else {
+		l, err := strconv.ParseUint(args["limit"], 10, strconv.IntSize)
+		if err != nil {
+			return fmt.Errorf("Invalid limit '%s', expected value >= 0", args["limit"])
+		}
+		limit = uint(l)
+	}
+
+	var offset uint
+	if args["offset"] != "" {
+		o, err := strconv.ParseUint(args["offset"], 10, strconv.IntSize)
+		if err != nil {
+			return fmt.Errorf("Invalid offset '%s', expected value >= 0", args["offset"])
+		}
+		offset = uint(o)
+	}
+
+	/* Save args. */
+	c.local = local
+	c.filter = proto.RequestFilter{
+		Type:   args["type"],
+		States: states,
+		User:   args["user"],
+
+		Since: since,
+		Until: until,
+
+		Limit:  limit,
+		Offset: offset,
+	}
+
+	return nil
+}
+
+func (c *Find) Run() error {
+	requests, err := c.ctx.RMClient.FindRequests(c.filter)
+	if err != nil {
+		return err
+	}
+	if c.ctx.Options.Debug {
+		app.Debug("requests: %#v", requests)
+	}
+
+	if c.ctx.Hooks.CommandRunResult != nil {
+		c.ctx.Hooks.CommandRunResult(requests, err)
+		return nil
+	}
+
+	if len(requests) == 0 {
+		return nil
+	}
+
+	/*
+	   ID                   REQUEST                                  USER      STATE     CREATED STARTED FINISHED JOBS
+	   -------------------- 1234567890123456789012345678901234567890 123456789 123456789 ------- ------- -------- *
+	*/
+	line := fmt.Sprintf("%%-%ds %%-%ds %%-%ds %%-%ds %%-%ds %%-%ds %%-%ds %%s\n",
+		findIdColLen, findReqColLen, findUserColLen, findStateColLen, findTimeColLen, findTimeColLen, findTimeColLen)
+
+	fmt.Fprintf(c.ctx.Out, line, "ID", "REQUEST", "USER", "STATE", "CREATED", "STARTED", "FINISHED", "JOBS")
+
+	timeConv := (time.Time).UTC
+	if c.local {
+		timeConv = (time.Time).Local
+	}
+
+	for _, r := range requests {
+		state, ok := proto.StateName[r.State]
+		if !ok {
+			state = proto.StateName[proto.STATE_UNKNOWN]
+		}
+
+		createdAt := timeConv(r.CreatedAt).Format(findTimeFmtStr)
+
+		startedAt := "N/A"
+		if r.StartedAt != nil {
+			startedAt = timeConv(*r.StartedAt).Format(findTimeFmtStr)
+		}
+
+		finishedAt := "N/A"
+		if r.FinishedAt != nil {
+			finishedAt = timeConv(*r.FinishedAt).Format(findTimeFmtStr)
+		}
+
+		jobs := fmt.Sprintf("%d / %d", r.FinishedJobs, r.TotalJobs)
+
+		fmt.Fprintf(c.ctx.Out, line,
+			SqueezeString(r.Id, findIdColLen, ".."),
+			SqueezeString(r.Type, findReqColLen, ".."),
+			SqueezeString(r.User, findUserColLen, ".."),
+			SqueezeString(state, findStateColLen, ".."),
+			createdAt, startedAt, finishedAt,
+			jobs)
+	}
+
+	return nil
+}
+
+func (c *Find) Cmd() string {
+	if len(c.ctx.Command.Args) > 0 {
+		return "find " + strings.Join(c.ctx.Command.Args, " ")
+	}
+	return "find"
+}
+
+func (c *Find) Help() string {
+	return fmt.Sprintf(`'spinc find [arg=value] [filter=value]' retrieves and filters requests.
```

Comment:
I think this command description can be improved. It should include what the command does by default without args given. Especially if `since` and `until` are not used, what requests are returned?